packages/@tailwindcss-cli/src/commands/build/index.ts

```diff
@@ -235,13 +249,59 @@ export async function handle(args: Result<ReturnType<typeof options>>) {
     // disable this behavior with `--watch=always`.
     if (args['--watch'] !== 'always') {
       process.stdin.on('end', () => {
+        cleanupWatchers()
         process.exit(0)
       })
     }
 
     // Keep the process running
     process.stdin.resume()
   }
+
+  await write(compiler.build(scanDirResult.candidates), args)
+
+  let end = process.hrtime.bigint()
+  eprintln(header())
+  eprintln()
+  eprintln(`Done in ${formatDuration(end - start)}`)
+}
+
+async function createWatchers(globs: GlobEntry[], handle: (files: string[]) => void) {
+  let watchers = disposables()
+  let files = new Set<string>()
+
+  let d = disposables()
+  function enqueueFlush() {
+    d.dispose()
+    d.queueMacrotask(() => {
+      handle(Array.from(files))
+      files.clear()
+    })
+  }
+
+  // Setup a watcher for every glob based on the `base` directory.
+  for (let glob of globs) {
+    // Globs with `!` are negated and should not require a dedicated watcher.
+    if (glob.glob[0] === '!') continue
+
+    let { unsubscribe } = await watcher.subscribe(glob.base, async (err, events) => {
+      if (err) {
+        console.error(err)
+        return
+      }
+
+      for (let event of events) {
+        if (event.type === 'delete') continue
+        files.add(event.path)
+      }
+
+      enqueueFlush()
+    })
+
+    watchers.add(unsubscribe)
+  }
+
+  return () => watchers.dispose()
```

Comment:
I think this function could use more documentation — there's a lot of little details and stuff in here that aren't really explained at the moment, like what `enqueueFlush` is for for example